test/index.test.js

```diff
@@ -49,31 +49,106 @@ describe('index', () => {
     plugin = new Plugin();
   });
 
-  describe('#constructor', () => {
+  describe('#methods', () => {
+    beforeEach(() => {
+      sinon.spy(plugin, '_prepareMessage');
+
+      plugin.loggers = [
+        { log: sinon.spy() },
+        { log: sinon.spy() }
+      ];
+    });
+
     it('should implement all methods listed in hooks', () => {
       Object.keys(plugin.hooks).forEach(k => {
         should(plugin[plugin.hooks[k]]).be.a.Function();
       });
     });
 
-    it('all hook methods should call the loggers', () => {
+    it('hook methods to their appropriate loggers', () => {
       Object.keys(plugin.hooks).forEach(k => {
-        plugin = new Plugin();
-
-        plugin.loggers = [
-          { log: sinon.spy() },
-          { log: sinon.spy() }
-        ];
-
         plugin[plugin.hooks[k]]('message', 'event');
+        should(plugin._prepareMessage).calledOnce().calledWith('message');
 
         plugin.loggers.forEach(logger => {
           should(logger.log).be.calledOnce();
-          should(logger.log).be.calledWith(k.replace(/^log:/, ''), 'event', 'message');
+          should(logger.log).be.calledWith(
+            k.replace(/^log:/, ''),
+            'event',
+            'message');
+
+          logger.log.resetHistory();
         });
+
+        plugin._prepareMessage.resetHistory();
```

Comment:
We don't need to reset in hooks before we instantiate new stubs for each test.

But we need to reset the spy here, because we're re-using it in a loop (starting line 69)